src/components/MovieCard/MovieCard.tsx

```diff
@@ -1,9 +1,9 @@
 import React from 'react';
-import { Movie } from '../../types/Movie';
+import { MovieData } from '../../types/MovieData';
 import './MovieCard.scss';
 
 type Props = {
-  movie: Movie,
+  movie: MovieData | null,
```

Comment:
You don't need to render the card if there is no movie 
```suggestion
  movie: MovieData,
```